mlir/lib/Dialect/Linalg/Transforms/ElementwiseOpFusion.cpp

```diff
@@ -816,17 +816,51 @@ validateDynamicDimExpansion(LinalgOp linalgOp,
 }
 
 // Create an expanded transpose op.
+// For bubbling a collapse : transpose(collapse_shape),
```

Comment:
I think the example is misleading. This is needed cause transpose requires the output to be identity indexing map, and that might not always happen when you do the reshape propagation (I think, I am not sure). 

Also you arent "bubbling" a collapse, you are sinking a collapse (bubbles go up, and here the collapse moves down).

Reply:
I fixed the bubbling/sinking naming error, about the permutation calculation. I think its more then needing the indexing maps a certain way. If you dont take the inverse of reassoc -> flatten -> inverse to get the expanded permutation in reshape as a consumer fusion, you just end up doing the wrong permutation on your input. Its because the reassociations are on the transposed layout but the expanded permutations we are looking for are on the untransposed layout. Am i missing something? Also if you have a concise way to describe the function below or would prefer to not have this comment/example as its confusing, happy to change/remove.

Reply:
Actually, I realized that I could use the dims from expansion info and make the logic
1. consistent across sinking collapse / bubbling expand
2. easier to follow
PTAL